### PR TITLE
Bugfix/deadline change has no save button

### LINKED ISSUE #418: Deadline updating doesn't work
# Bug report

## Expected Behavior
When editing a deadline of a node in the format editor, the save icon shows up.

## Actual Behavior
When editing a deadline of a node in the format editor, the save icon doesn't show up.

## Steps to Reproduce the Problem
1. Login as a teacher
2. Go to a assignment and edit it.
3. Click on one of the nodes on the left.
4. Edit the deadline.
5. See that the save icon doesn't show up in the bottem right corner.

# Feature request

## Description
Make it so that the change of the deadline in a node is detected and triggers the save button to come up.

### PR DESCRIPTION
**Pull request by Jeroentetje3**
## Summary of changes
- Fixes #418

### CODE REVIEW

File: src/vue/src/components/format/FormatEditSelectTemplateCard.vue [hunk 1]
```diff
@@ -47,7 +47,8 @@
             </b-col>
         </b-row>
         <h2 class="field-heading">Deadline</h2>
-        <flat-pickr class="theme-input multi-form full-width" v-model="currentPreset.deadline" :config="$root.flatPickrTimeConfig"/>
+        <flat-pickr class="theme-input multi-form full-width" v-model="currentPreset.deadline"
```
Comment (mjvkeulen) [inline src/vue/src/components/format/FormatEditSelectTemplateCard.vue]: We're not super consistent with our HTML style. However, if a line becomes too long each property is separated by newlines. Not just the one that goes over. 

image

**Comment by engelhamer** [comment]
I think you should also add this to the date picker for the assignment due date etc.

File: src/vue/src/components/format/FormatEditSelectTemplateCard.vue [hunk 2]
```diff
@@ -46,8 +46,13 @@
...
             </b-col>
         </b-row>
-        <h2 class="field-heading">Deadline</h2>
-        <flat-pickr class="theme-input multi-form full-width" v-model="currentPreset.deadline" :config="$root.flatPickrTimeConfig"/>
+        <h2 class="field-heading">Deadline bla</h2>
+        <flat-pickr
+            class="theme-input multi-form full-width"
+            v-model="currentPreset.deadline"
+            :config="$root.flatPickrTimeConfig"
+            @on-change="$emit('changed')"
```
Comment (Larspolo) [inline src/vue/src/components/format/FormatEditSelectTemplateCard.vue]: inconsistant `on-`

File: src/vue/src/components/format/FormatEditSelectTemplateCard.vue [hunk 3]
```diff
@@ -64,7 +69,8 @@
         </div>
         <div v-else-if="currentPreset.type === 'p'">
             <h2 class="field-heading">Point Target</h2>
-            <b-input type="number" class="theme-input" v-model="currentPreset.target" placeholder="Amount of points" @change="$emit('changed')"/>
+            <b-input type="number" class="theme-input" v-model="currentPreset.target" placeholder="Amount of points"
+                @change="$emit('changed')"/>
```
Comment (Larspolo) [inline src/vue/src/components/format/FormatEditSelectTemplateCard.vue]: inconsistant `on-`

File: src/vue/src/components/format/FormatEditSelectTemplateCard.vue [hunk 4]
```diff
@@ -46,8 +46,13 @@
                 </b-card>
             </b-col>
         </b-row>
-        <h2 class="field-heading">Deadline</h2>
-        <flat-pickr class="theme-input multi-form full-width" v-model="currentPreset.deadline" :config="$root.flatPickrTimeConfig"/>
+        <h2 class="field-heading">Deadline bla</h2>
```
[the remark this answers is not in the document]
Comment (engelhamer) [reply src/vue/src/components/format/FormatEditSelectTemplateCard.vue]: Still there.

File: src/vue/src/components/format/FormatEditSelectTemplateCard.vue (hunk 2 above)
Comment (Airgetfrog) [reply src/vue/src/components/format/FormatEditSelectTemplateCard.vue]: This is caused by the [vue-flatpickr event naming](https://www.npmjs.com/package/vue-flatpickr-component?activeTab=readme#events), and is not an error.

**Comment by engelhamer** [comment]
Outdated. This will be fixed in another pull request.

**Pull request closed without merging**